pandas/core/arrays/arrow/dtype.py

```diff
@@ -321,3 +326,27 @@ def __from_arrow__(self, array: pa.Array | pa.ChunkedArray):
         array_class = self.construct_array_type()
         arr = array.cast(self.pyarrow_dtype, safe=True)
         return array_class(arr)
+
+    def _maybe_promote(self, item: Any) -> tuple[DtypeObj, Any]:
```

Comment:
Shouldn't this be implemented this on `ExtensionDtype` too?

Reply:
it is https://github.com/pandas-dev/pandas/pull/53106/files#diff-2e345cb935fc3bae33ad469d4187e9607f679f19e30afb6d764f7c7c4f32639dR394

Reply:
Ah gotcha. I didn't notice that. Would be good for the base function to have a docstring (as you mentioned in the OP) so it's clear what this method is supposed to do